experimental/Matrix/matrix.g

```diff
@@ -0,0 +1,139 @@
+                       IsPosInt and IsSmallIntRep ],
+    function( m, i, j )
+      return (m!.m)[i,j];
+    end );
+    
+InstallOtherMethod( \*, [IsJuliaMatrixRep, IsJuliaMatrixRep], function( m1, m2 )
+        return MakeJuliaMatrixRep(m1!.m * m2!.m);
+    end );
+    
+InstallOtherMethod( \/, [IsJuliaMatrixRep, IsJuliaMatrixRep], function( m1, m2 )
+        return m1 * Inverse(m2);
+    end);
+    
+InstallMethod( InverseMutable, [IsJuliaMatrixRep], m -> MakeJuliaMatrixRep(InverseOp(m!.m)));
+
+InstallMethod( InverseImmutable, [IsJuliaMatrixRep], m -> MakeImmutable(MakeJuliaMatrixRep(InverseOp(m!.m))));
+
+InstallOtherMethod( AdditiveInverseOp, [IsJuliaMatrixRep], m -> MakeJuliaMatrixRep(AdditiveInverseOp(m!.m)));
+
+InstallOtherMethod(One, [IsJuliaMatrixRep], m -> MakeJuliaMatrixRep(OneOp(m!.m)));
+
+InstallOtherMethod(Zero, [IsJuliaMatrixRep], m -> MakeJuliaMatrixRep(ZeroOp(m!.m)));
+
+InstallOtherMethod(\+, [IsJuliaMatrixRep, IsJuliaMatrixRep], function( m1, m2 )
+        return MakeJuliaMatrixRep(m1!.m + m2!.m);
+    end);
+    
+InstallOtherMethod(\-, [IsJuliaMatrixRep, IsJuliaMatrixRep], function( m1, m2 )
+        return MakeJuliaMatrixRep(m1!.m - m2!.m);
+    end);
+
+InstallOtherMethod(\^, [IsJuliaMatrixRep, IsInt and IsSmallIntRep], function( m1, k )
+        if k < 0 then
+            return MakeJuliaMatrixRep( (Inverse(m1)!.m)^(-k) );
+        else
+            return MakeJuliaMatrixRep( (m1!.m)^k );
+        fi;
+    end);
+    
+InstallOtherMethod(\=, [IsJuliaMatrixRep, IsJuliaMatrixRep], function( m1, m2 )
+        return m1!.m = m2!.m;
+    end);
+    
+InstallOtherMethod(Display, [IsJuliaMatrixRep], 5000, function( m )
+    local i,j;
+        Print("[ ");
+        for i in [1..NumberRows(m)] do
+            Print("[ ");
+            for j in [1..NumberColumns(m)] do
+                Print(m[i,j]);
+                if (j < NumberColumns(m)) then
+                    Print(", ");
+                fi;
+            od;
+            if (i < NumberRows(m)) then
+                Print(" ] \n");
+                Print("  ");
+            else
+                Print(" ] ] \n");
+            fi;
+        od;
+    end);
+
+InstallOtherMethod(Unpack, [IsJuliaMatrixRep], function(m)
+    local v, i, j;
+    v := EmptyPlist(NumberRows(m)*NumberColumns(m));
+    for i in [1..NumberRows(m)] do
+        for j in [1..NumberColumns(m)] do
+            v[j+NumberColumns(m)*(i-1)] := m[i,j];
+        od;
+    od;
+    return v;
+end);
+
+# TODO: Not working. JuliaMatrixRep is immutable
+InstallOtherMethod( \[\,\]\:\=,
+    [ IsJuliaMatrixRep and IsMutable, IsPosInt and IsSmallIntRep,
+                       IsPosInt and IsSmallIntRep, IsObject ],
+    function( m, i, j, val )
+    local mat;
+        mat := m!.m;
+        mat[i,j] := val;
+    end );
+
+# TODO: Not working
+InstallOtherMethod(\<, [IsJuliaMatrixRep, IsJuliaMatrixRep], function( m1, m2 )
+        return m1!.m < m2!.m;
+    end);
```

Comment:
Move this next to the `\=` method anyway?

Also, in how far is this "not working"?

Reply:
Done. I collected the not working methods at the end.